modules/fitsfilereader/src/wsafitshelper.cpp

```diff
@@ -0,0 +1,166 @@
+                                                         size_t layerIndex, float minMax)
+{
+    try {
+        std::unique_ptr<FITS> file = std::make_unique<FITS>(path.string(), Read, true);
+        if (!file.get()) {
+            LERROR(std::format(
+                "Failed to open, therefor removing file {}", path.string()
+            ));
+            std::filesystem::remove(path);
+            return nullptr;
+        }
+        // Convert fits path with fits-file-reader functions
+        const std::shared_ptr<ImageData<float>> fitsValues =
+            readImageInternal<float>(file->pHDU());
+        int layerSize = fitsValues->width * fitsValues->height;
+
+        int nLayers = fitsValues->contents.size() / layerSize;
+        if (layerIndex > nLayers -1) {
+            LERROR(
+                "Chosen layer in fits file is not supported. Index to high. ",
+                "First layer chosen instead"
+            );
+            layerIndex = 0;
+        }
+
+        std::valarray<float> layerValues =
+            fitsValues->contents[std::slice(layerIndex*layerSize, layerSize, 1)];
+
+        const float maxValue = minMax;
+        const float minValue = -minMax;
+        float* imageData = new float[layerValues.size()];
+        std::vector<glm::vec3> rgbLayers;
+        for (size_t i = 0; i < layerValues.size(); i++) {
+            // normalization
+            float normalizedValue = (layerValues[i] - minValue) / (maxValue - minValue);
+            // clamping causes overexposure above and below max and min values
+            // intentionally as desired by Nick Arge from WSA
+            normalizedValue = std::clamp(normalizedValue, 0.f, 1.f);
+
+            imageData[i] = normalizedValue;
+        }
+
+        // Create texture from imagedata
+        auto texture = std::make_unique<ghoul::opengl::Texture>(
+            imageData,
+            glm::size3_t(fitsValues->width, fitsValues->height, 1),
+            GL_TEXTURE_2D,
+            ghoul::opengl::Texture::Format::Red,
+            GL_RED,
+            GL_FLOAT
+        );
+        // Tell it to use the single color channel as grayscale
+        convertTextureFormat(*texture, ghoul::opengl::Texture::Format::RGB);
+        texture->uploadTexture();
+        return texture;
+    }
+    catch (const CCfits::FitsException& e) {
+        LERROR(std::format(
+            "Failed to open fits file '{}'. '{}'", path.string(), e.message()
+        ));
+        std::filesystem::remove(path);
+        return nullptr;
+    }
+    catch (const std::exception& e) {
+        LERROR(std::format(
+            "Failed to open fits file '{}'. '{}'", path, e.what()
+        ));
+        std::filesystem::remove(path);
+        return nullptr;
+    }
+    catch (...) {
+        LERROR(std::format(
+            "Unknown exception caught for file '{}'", path
+        ));
+        std::filesystem::remove(path);
+        return nullptr;
+    }
```

Comment:
What are the reasons for these? They are both extremely wide nets to catch exceptions and there should be a good reason for them